### PR TITLE
chore(compass-sidebar): convert to new plugin interface COMPASS-7443

### PR DESCRIPTION
**Pull request by addaleax**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-sidebar/src/index.ts [hunk 1]
```diff
@@ -1,25 +1,63 @@
...
+import type { MongoDBInstance } from 'mongodb-instance-model';
 
-/**
- * Activate all the components in the Sidebar package.
- * @param {Object} appRegistry - The Hadron appRegisrty to activate this plugin with.
- **/
-function activate(appRegistry: AppRegistry) {
-  appRegistry.registerComponent('Sidebar.Component', SidebarPlugin);
-  appRegistry.registerStore('Sidebar.Store', SidebarStore);
+function activate() {
```
Comment (Anemy) [inline packages/compass-sidebar/src/index.ts:10]: Can we remove this `activate` and `deactivate` now? Or are they supposed to be kept around for the time being?

Digging this new `registerHadronPlugin` usage !

File: packages/compass-sidebar/src/components/sidebar.tsx [hunk 2]
```diff
@@ -219,6 +220,7 @@ const MappedSidebar = connect(mapStateToProps, {
   updateAndSaveConnectionInfo,
   toggleIsGenuineMongoDBVisible,
   setIsExpanded,
+  setConnectionIsCSFLEEnabled,
```
Comment (Anemy) [inline packages/compass-sidebar/src/components/sidebar.tsx:223]: Looks like we still have the `sidebar-toggle-csfle-enabled` event being emitted here, should we remove that and use this function?

File: packages/compass-sidebar/src/index.ts (hunk 1 above)
Comment (addaleax) [reply packages/compass-sidebar/src/index.ts:10]: Right now, we are still loading all of these plugins into the PluginManager in the core compass package, and that interface expects these methods to be available. But that should be something we can get rid off after doing the rest of the conversion work :)

File: packages/compass-sidebar/src/components/sidebar.tsx (hunk 2 above)
Comment (addaleax) [reply packages/compass-sidebar/src/components/sidebar.tsx:218]: Yeah, thanks for catching this … :see_no_evil:

**Review by gribnoysup — APPROVED** [review]

**Pull request merged**